test/steward_multi_patronage.js

```diff
@@ -0,0 +1,139 @@
+    //////////////////////////////////////////////////
+    await time.increase(time.duration.minutes(10));
+    // await waitTillBeginningOfSecond()
+
+    // Buy a 2nd token
+    const buyToken2Tx = await steward.buy(testTokenId2, web3.utils.toWei('2', 'ether'), { from: accounts[2], value: web3.utils.toWei('1', 'ether') });
+    const buyToken2BlockTime = (await web3.eth.getBlock(buyToken2Tx.receipt.blockNumber)).timestamp
+    const lastCollectedUserT20 = await steward.timeLastCollectedUser.call(accounts[2])
+    const priceOfToken2 = await steward.price.call(testTokenId2)
+    assert.equal(buyToken2BlockTime.toString(), lastCollectedUserT20.toString())
+
+    const patronDepositAfter20min = await steward.deposit.call(accounts[2]);
+    const patronDepositCalculatedAfter20min = await steward.depositAbleToWithdraw.call(accounts[2]);
+    const expectedPatronage10MinToken1 = multiPatronageCalculator('600',
+      [{ patronageNumerator: testToken1.patronageNumerator.toString(), price: priceOfToken1.toString() }])
+
+    assert.equal(patronDepositAfter20min.toString(), patronDepositAfter10min.sub(expectedPatronage10MinToken1).add(ether('1')).toString());
+    assert.equal(patronDepositCalculatedAfter20min.toString(), patronDepositAfter10min.add(ether('1')).sub(expectedPatronage10MinToken1).toString());
+
+    /////////////////// TIME = 30 ////////////////////
+    //////////////////////////////////////////////////
+    //////////////////////////////////////////////////
+    await time.increase(time.duration.minutes(10));
+    await waitTillBeginningOfSecond()
+
+    await steward._collectPatronage(testTokenId1);
+
+    const patronDepositAfter30min = await steward.deposit.call(accounts[2]);
+    const patronDepositCalculatedAfter30min = await steward.depositAbleToWithdraw.call(accounts[2]);
+    const expectedPatronageMulti = multiPatronageCalculator('601',
+      [{ patronageNumerator: testToken1.patronageNumerator.toString(), price: priceOfToken1.toString() },
+      { patronageNumerator: testToken2.patronageNumerator.toString(), price: priceOfToken2.toString() }])
+
+    assert.equal(patronDepositAfter20min.toString(), patronDepositAfter30min.add(expectedPatronageMulti).toString());
+    assert.equal(patronDepositCalculatedAfter30min.toString(), patronDepositAfter30min.toString());
+
+    const benefactorFundsT30 = await steward.benefactorFunds.call(accounts[8])
+    const balTrack = await balance.tracker(accounts[8]);
+    await steward.withdrawBenefactorFundsTo(accounts[8])
+    const balanceChangePatron1 = await balTrack.delta()
+
+    const expectedTotalPatronageT30Token1 = multiPatronageCalculator('1801',
+      [{ patronageNumerator: testToken1.patronageNumerator.toString(), price: priceOfToken1.toString() }])
+    assert.equal(benefactorFundsT30.toString(), expectedTotalPatronageT30Token1.toString())
+    assert.equal(balanceChangePatron1.toString(), expectedTotalPatronageT30Token1.toString())
+  });
+});
```

Comment:
Also add a test to check the patronage amount received by account[9]  is correct for token 2? 
const benefactorFundsT30token2 = await steward.benefactorFunds.call(accounts[9])
const expectedTotalPatronageT30Token2 = 
multiPatronageCalculator('601',
      [{ patronageNumerator: testToken2.patronageNumerator.toString(), price: priceOfToken2.toString() }])
assert.equal(benefactorFundsT30token2.toString(), expectedTotalPatronageT30Token2.toString())

Reply:
Sure, we can. Maybe a new section, and add some more time.